testsuite/src/main/java/io/netty/testsuite/transport/socket/SocketHalfClosedTest.java

```diff
@@ -414,19 +416,26 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) {
 
         @Override
         public void channelInactive(ChannelHandlerContext ctx) {
-            checkPrematureClose();
+            checkPrematureClose(ctx);
         }
 
         @Override
         public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
             ctx.close();
-            checkPrematureClose();
+            checkPrematureClose(ctx);
         }
 
-        private void checkPrematureClose() {
+        private void checkPrematureClose(ChannelHandlerContext ctx) {
             if (bytesRead < expectedBytes || !seenOutputShutdown) {
-                causeRef.set(new IllegalStateException("leader premature close"));
-                doneLatch.countDown();
+                if (ctx.channel() instanceof OioSocketChannel && seenOutputShutdown
+                        && PlatformDependent.javaVersion() >= 11) {
```

Comment:
Perhaps we should check the Java version before any other conditions because it's more likely to fail fast on most pre-11 deployments?

Reply:
Ah, missed that it's a test class. No worries, then!